libs/aws/tests/test_aws.py

```diff
@@ -111,3 +111,47 @@ def test_get_queue_attributes(aws_env):
     for att in valid_attributes:
         att_returned = get_queue_attributes(queue_name="queue1", attribute=att)
         assert att_returned.get(att)
+
+
+@mock_sqs
+def test_get_queue_attributes(aws_env):
+    resource = boto3.resource("sqs")
+
+    resource.create_queue(QueueName="queue1")
+
+    valid_attributes = [
+        'ApproximateNumberOfMessages',
+        'ApproximateNumberOfMessagesDelayed',
+        'ApproximateNumberOfMessagesNotVisible',
+        'CreatedTimestamp',
+        'DelaySeconds',
+        'LastModifiedTimestamp',
+        'MaximumMessageSize',
+        'MessageRetentionPeriod',
+        'QueueArn',
+        'ReceiveMessageWaitTimeSeconds',
+        'VisibilityTimeout'
+    ]
+
+    all_attributes = get_queue_attributes(queue_name="queue1")
+
+    assert len(all_attributes) == len(valid_attributes)
+
+    for att in valid_attributes:
+        att_returned = get_queue_attributes(queue_name="queue1", attribute=att)
+        assert att_returned.get(att)
```

Comment:
Asserting that an object exists isn't really enough.

Maybe assert is not None? Or that the attribute is what you set it to?

Reply:
Sure

Reply:
**is not None** it is because the values are created by the library when the queue is created. So it might change in the future.